### ISSUE TITLE
Tree - Random node placement between subsequent calls, and orphans between siblings

### ISSUE THREAD

**Issue by jfstephe** [OpenKieler/elkjs#39]
Objective
========
I'm trying to get a tree hierarchy of nodes (think family tree) and be able to add to the graph new nodes over time (when people are born for example).

Background:
========

I started using the MrTree algorithm but the edges kept crossing and I couldn't seem to un-cross them and the placement of nodes trees was in a wrapped manner not horizontal as I require.

![image](https://user-images.githubusercontent.com/3627080/42213372-770f9926-7eb0-11e8-8a6d-7faf810aaaf3.png)

So I ended up with the config below (some settings don't seem to do much e.g hierarchicalSweepiness)...
```
'org.eclipse.elk.layered.crossingMinimization.hierarchicalSweepiness': 1,
'org.eclipse.elk.hierarchyHandling': 'INCLUDE_CHILDREN',
'org.eclipse.elk.debugMode': true,
'org.eclipse.elk.layered.crossingMinimization.strategy': 'LAYER_SWEEP',
'org.eclipse.elk.direction': 'DOWN',
'org.eclipse.elk.layered.layering.strategy': 'LONGEST_PATH',
'org.eclipse.elk.layered.nodePlacement.strategy': 'BRANDES_KOEPF',
'org.eclipse.elk.zoomToFit': true,
'org.eclipse.elk.alignment': 'TOP'

```
The Problems
=========
I've got 2 issues, mentioned in the screenshot below. This shows orphan nodes being added repeatedly over time.

![image](https://user-images.githubusercontent.com/3627080/42213038-a77435fa-7eaf-11e8-9286-138079d3701d.png)

In short these are:
1) Placement of new nodes between siblings nodes of a tree isn't acceptable
2) The random arrangement of nodes between layout calls

I think I'm missing something, I've tried using the INTERACTIVE for node placement but again I have edge crossing issues. Any suggestions would be greatly appreciated.

**Comment by uruuru** [comment]
The layered algorithm is designed to provide a _stable_ layout of an incrementally changing graph. There are algorithms out there (at least in the literature) that tackle this problem for general graphs and for trees in particular. 

Your second point (random arrangement of nodes between layout calls) shouldn't happen unless you change something in the graph's structure (or there's a bug but I haven't seen this behavior so far). 

You could try one further thing though: 
 * set [`crossingMinimization.semiInteractive`](https://www.eclipse.org/elk/reference/options/org-eclipse-elk-layered-crossingminimization-semiinteractive.html): `true` and `crossingMinimization.strategy: LAYER_SWEEP`. 
 * then prior to a new layout, set each node's [`elk.position`](https://www.eclipse.org/elk/reference/options/org-eclipse-elk-position.html) **option** to it's current position. 
This _may_ help to preserve the horizontal order better. Note, however, that you'll eventually end up with edge crossings that you feel could be avoided ...

**Comment by uruuru** [comment]
Also, for the record, you can find the layout options that are supported by an algorithm on the corresponding documentation page [here](). 

In your case, only the following options have an effect: 
```
'org.eclipse.elk.layered.crossingMinimization.strategy': 'LAYER_SWEEP',
'org.eclipse.elk.direction': 'DOWN',
'org.eclipse.elk.layered.layering.strategy': 'LONGEST_PATH',
'org.eclipse.elk.layered.nodePlacement.strategy': 'BRANDES_KOEPF',
```
